api/v1beta1/common_types.go

```diff
@@ -53,6 +53,10 @@ const (
 	//   will not be completed until the annotation is removed and all MachineDeployments are upgraded.
 	ClusterTopologyDeferUpgradeAnnotation = "topology.cluster.x-k8s.io/defer-upgrade"
 
+	// ClusterTopologyUpgradeConcurrencyAnnotation can be set on a Classy Cluster to define the maximum concurrency while
+	// upgrading MachineDeployments of a Classy Cluster.
```

Comment:
```suggestion
	// ClusterTopologyUpgradeConcurrencyAnnotation can be set as top-level annotation on the Cluster object of 
	// a classy Cluster to define the maximum concurrency while upgrading MachineDeployments.
```
nit: Let's explicitly mention that this annotation should be set on the Cluster resource. (just because we have two similar annotations just right above which are set in a different place)

(I think it's enough to have "of a classy Cluster" just once)